pmm/v3/pmm3-submodules.groovy

```diff
@@ -63,26 +78,31 @@ pipeline {
                         echo $pmm_ui_tests_branch > pmmUITestBranch
                         echo $pmm_ui_tests_commit > pmmUITestsCommitSha
                     else
-                        # This build method requires feature branches to be manually set in .gitmodules
-                        git submodule update --init --jobs 10
+                        sudo rm -rf results tmp || :
+                        git reset --hard
+                        git clean -fdx
+                        git submodule foreach --recursive git reset --hard
+                        git submodule foreach --recursive git clean -fdx
                         git submodule status
-                        # Define variables
-                        pmm_commit=$(git submodule status | grep 'sources/pmm/src' | awk -F ' ' '{print $1}')
-                        echo $pmm_commit > apiCommitSha
-                        pmm_branch=$(git config -f .gitmodules submodule.pmm.branch)
-                        echo $pmm_branch > apiBranch
-                        pmm_url=$(git config -f .gitmodules submodule.pmm.url)
-                        echo $pmm_url > apiURL
-                        pmm_qa_branch=$(git config -f .gitmodules submodule.pmm-qa.branch)
+                        export commit_sha=$(git submodule status | grep 'pmm-managed' | awk -F ' ' '{print $1}')
+                        export api_tests_commit_sha=$(git submodule status | grep 'sources/pmm/src' | awk -F ' ' '{print $1}')
+                        export api_tests_branch=$(git config -f .gitmodules submodule.pmm.branch)
+                        export api_tests_url=$(git config -f .gitmodules submodule.pmm.url)
+                        echo $api_tests_commit_sha > apiCommitSha
+                        echo $api_tests_branch > apiBranch
+                        echo $api_tests_url > apiURL
+                        cat apiBranch
+                        cat apiURL
+                        export pmm_qa_commit_sha=$(git submodule status | grep 'pmm-qa' | awk -F ' ' '{print $1}')
+                        export pmm_qa_branch=$(git config -f .gitmodules submodule.pmm-qa.branch)
```

Comment:
How can we justify so many changes to this pipeline? It has been refactored on purpose, and now it seems like we restore the old version. Is there any reason why we should?